### PR TITLE
Update vision svc organization to be by model

### PR DESCRIPTION
**Pull request by npentrel**
# Description

Please include a summary of the change and what is does. Please also include relevant motivation and context.

### CODE REVIEW

File: docs/ml/vision/_index.md [hunk 1]
```diff
@@ -18,23 +22,83 @@ While the camera component lets you access what your machine's camera sees, the
...
+<div class="td-max-width-on-larger-screens">
+  <div class="alignright" >
+    {{< imgproc alt="A white dog with a bounding box around it labeled 'Dog: 0.71'" src="/ml/vision/dog-detector.png" resize="300x" declaredimensions=true >}}
+  </div>
+</div>
+
+_2D Object Detection_ is the process of taking a 2D image from a camera and identifying and drawing a box around the distinct "objects" of interest in the scene.
+Any camera that can return 2D images can use 2D object detection.
+
+The service provides different types of detectors, both based on heuristics and machine learning, so that you can create, register, and use detectors for any object you may need to identify.
```
Comment (sguequierre) [inline docs/ml/vision/_index.md:40]: (opt) It sounds a little off to me to say "The service provides" when it is different versions of the service. Maybe rephrase like "The service is provided as"?

File: docs/ml/vision/mlmodel.md [hunk 2]
```diff
@@ -326,6 +532,17 @@ To learn more about how to use classification, see the [Go SDK docs](https://pkg
 {{% /tab %}}
 {{< /tabs >}}
 
+{{% /tab %}}
+{{< /tabs >}}
+
 {{% alert title="Tip" color="tip" %}}
-To see more code examples of how to use Viam's vision service, see [our example repo](https://github.com/viamrobotics/vision-service-examples).
+To see more code examples of how to use Viam's Vision Service, see [our example repo](https://github.com/viamrobotics/vision-service-examples).
```
Comment (sguequierre) [inline docs/ml/vision/mlmodel.md:539]:
```suggestion
To see more code examples of how to use Viam's vision service, see [our example repo](https://github.com/viamrobotics/vision-service-examples).
```

File: docs/ml/vision/_index.md [hunk 3]
```diff
@@ -18,23 +22,83 @@ While the camera component lets you access what your machine's camera sees, the
 
 Currently, the vision service supports the following kinds of operations:
 
-{{< cards >}}
-{{% card link="/ml/vision/detection/" %}}
-{{% card link="/ml/vision/classification/" %}}
-{{% card link="/ml/vision/segmentation" %}}
-{{< /cards >}}
+- [Detections](#detections)
+- [Classificaitons](#classifications)
```
Comment (sguequierre) [inline docs/ml/vision/_index.md:26]:
```suggestion
- [Classifications](#classifications)
```

File: docs/ml/vision/obstacles_depth.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+weight: 20
+type: "docs"
+description: "A segmenter for depth cameras that returns the perceived obstacles as a set of 3-dimensional bounding boxes, each with a Pose as a vector."
+tags: ["vision", "computer vision", "CV", "services", "segmentation"]
+# SMEs: Bijan, Khari
+---
+
+_Changed in [RDK v0.2.36 and API v0.1.118](/appendix/changelog/#april-2023)_
+
+## Configure an `obstacles_depth` segmenter
```
Comment (sguequierre) [inline docs/ml/vision/obstacles_depth.md:13]:
```suggestion
```
This one is the only one that has that heading

File: docs/ml/vision/_index.md [hunk 5]
```diff
@@ -18,23 +22,83 @@ While the camera component lets you access what your machine's camera sees, the
...
+
+## Segmentations
+
+_3D Object Segmentation_ is the process of separating and returning a list of the identified "objects" from a 3D scene.
+The "objects" are usually a list of point clouds with associated metadata, like the label, the 3D bounding box, and center coordinates of the object.
+
+3D object segmentation is useful for obstacle detection.
+See our guide [Navigate with a Rover Base](/tutorials/services/navigate-with-rover-base/#next-steps-automate-obstacle-detection) for an example of automating obstacle avoidance with 3D object segmentation for obstacle detection.
+
+Any camera that can return 3D pointclouds can use 3D object segmentation.
```
Comment (sguequierre) [inline docs/ml/vision/_index.md:70]:
```suggestion
Any camera that can return 3D pointclouds can use 3D object segmentation.

**API Methods Supported:**
  - [GetObjectPointClouds()](#getobjectpointclouds)
```

(opt) Maybe with different formatting -- but I would suggest adding something like this here and likewise for classifications/segmentations to make that easy for users to see

File: docs/ml/vision/_index.md [hunk 6]
```diff
@@ -18,23 +22,83 @@ While the camera component lets you access what your machine's camera sees, the
...
+
+## Supported Models
+
+To use your motor with Viam, check whether one of the following [built-in models](#built-in-models) or [modular resources](#modular-resources) supports your motor.
+
+{{< readfile "/static/include/create-your-own-mr.md" >}}
+
+### Built-in models
+
+For configuration information, click on the model name:
```
Comment (sguequierre) [inline docs/ml/vision/_index.md:80]: (opt) Adding a column like `| Type | `  ` | _detector_ | ` would be helpful IMO. For most it is obvious but I am confused on those that are seemingly both segmenters/detectors (are those separate things?)

**Review by sguequierre — CHANGES_REQUESTED** [review]
Most opt, great changes!

File: docs/ml/vision/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/ml/vision/_index.md:40]: Hmm I don't like the passive voice. How do you feel about "you can use the service as " or " you can use the service with"?

File: docs/ml/vision/obstacles_depth.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+linkTitle: "obstacles_depth"
+weight: 20
+type: "docs"
+description: "A segmenter for depth cameras that returns the perceived obstacles as a set of 3-dimensional bounding boxes, each with a Pose as a vector."
+tags: ["vision", "computer vision", "CV", "services", "segmentation"]
+# SMEs: Bijan, Khari
+---
+
+_Changed in [RDK v0.2.36 and API v0.1.118](/appendix/changelog/#april-2023)_
+
```
Comment (npentrel) [inline docs/ml/vision/obstacles_depth.md:12]:
```suggestion
```

File: docs/ml/vision/_index.md (hunk 6 above)
Comment (npentrel) [reply docs/ml/vision/_index.md:80]: You mean the one that uses a detector to turn it into segmenter output. It is a segmenter. I can't think how to make the description better

File: docs/ml/vision/_index.md [hunk 8]
```diff
@@ -18,23 +22,87 @@ While the camera component lets you access what your machine's camera sees, the
...
+Any camera that can return 2D images can use 2D image classification.
+
+The class labels used for classification vary and depend on the machine learning model and how it was trained.
+
+The returned classifications consist of the image's class label and confidence score.
+
+- `class_name` (string): specifies the label of the found object.
+- `confidence` (float): specifies the confidence of the assigned label.
+  Between `0.0` and `1.0`, inclusive.
+
```
Comment (npentrel) [inline docs/ml/vision/_index.md:61]:
```suggestion

**Supported API methods:**

  - [GetClassifications()](#getclassifications)
  - [GetClassificationsFromCamera()](#getclassificationsfromcamera)

```

File: docs/ml/vision/_index.md [hunk 9]
```diff
@@ -18,23 +22,87 @@ While the camera component lets you access what your machine's camera sees, the
...
+The service provides different types of detectors, both based on heuristics and machine learning, so that you can create, register, and use detectors for any object you may need to identify.
+
+The returned detections consist of the bounding box around the identified object, as well as its label and confidence score:
+
+- `x_min`, `y_min`, `x_max`, `y_max` (int): specify the bounding box around the object.
+- `class_name` (string): specifies the label of the found object.
+- `confidence` (float): specifies the confidence of the assigned label.
+  Between `0.0` and `1.0`, inclusive.
+
```
Comment (npentrel) [inline docs/ml/vision/_index.md:48]:
```suggestion

**Supported API methods:**

  - [GetDetections()](#getdetections)
  - [GetDetectionsFromCamera()](#getdetectionsfromcamera)

```

File: docs/ml/vision/_index.md [hunk 10]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+The "objects" are usually a list of point clouds with associated metadata, like the label, the 3D bounding box, and center coordinates of the object.
+
+3D object segmentation is useful for obstacle detection.
+See our guide [Navigate with a Rover Base](/tutorials/services/navigate-with-rover-base/#next-steps-automate-obstacle-detection) for an example of automating obstacle avoidance with 3D object segmentation for obstacle detection.
+
+Any camera that can return 3D pointclouds can use 3D object segmentation.
+
+**Supported API methods:**
+
+  - [GetObjectPointClouds()](#getobjectpointclouds)
```
Comment (npentrel) [inline docs/ml/vision/_index.md:84]:
```suggestion
- [GetObjectPointClouds()](#getobjectpointclouds)
```

File: docs/ml/vision/_index.md [hunk 11]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+
+The returned classifications consist of the image's class label and confidence score.
+
+- `class_name` (string): specifies the label of the found object.
+- `confidence` (float): specifies the confidence of the assigned label.
+  Between `0.0` and `1.0`, inclusive.
+
+**Supported API methods:**
+
+  - [GetClassifications()](#getclassifications)
```
Comment (npentrel) [inline docs/ml/vision/_index.md:69]:
```suggestion
- [GetClassifications()](#getclassifications)
```

File: docs/ml/vision/_index.md [hunk 12]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+  - [GetClassificationsFromCamera()](#getclassificationsfromcamera)
```
Comment (npentrel) [inline docs/ml/vision/_index.md:70]:
```suggestion
- [GetClassificationsFromCamera()](#getclassificationsfromcamera)
```

File: docs/ml/vision/_index.md [hunk 13]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+The returned detections consist of the bounding box around the identified object, as well as its label and confidence score:
+
+- `x_min`, `y_min`, `x_max`, `y_max` (int): specify the bounding box around the object.
+- `class_name` (string): specifies the label of the found object.
+- `confidence` (float): specifies the confidence of the assigned label.
+  Between `0.0` and `1.0`, inclusive.
+
+**Supported API methods:**
+
+  - [GetDetections()](#getdetections)
```
Comment (npentrel) [inline docs/ml/vision/_index.md:51]:
```suggestion
- [GetDetections()](#getdetections)
```

File: docs/ml/vision/_index.md [hunk 14]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+  - [GetDetectionsFromCamera()](#getdetectionsfromcamera)
```
Comment (npentrel) [inline docs/ml/vision/_index.md:52]:
```suggestion
- [GetDetectionsFromCamera()](#getdetectionsfromcamera)
```

File: docs/ml/vision/_index.md [hunk 15]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+<div class="td-max-width-on-larger-screens">
+  <div class="alignright" >
+    {{< imgproc alt="A white dog with a bounding box around it labeled 'Dog: 0.71'" src="/ml/vision/dog-detector.png" resize="300x" declaredimensions=true >}}
+  </div>
+</div>
+
+_2D Object Detection_ is the process of taking a 2D image from a camera and identifying and drawing a box around the distinct "objects" of interest in the scene.
+Any camera that can return 2D images can use 2D object detection.
+
+The service provides different types of detectors, both based on heuristics and machine learning, so that you can create, register, and use detectors for any object you may need to identify.
```
Comment (npentrel) [inline docs/ml/vision/_index.md:40]:
```suggestion
You can use different types of detectors, both based on heuristics and machine learning, for any object you may need to identify.
```

File: docs/ml/vision/_index.md [hunk 16]
```diff
@@ -18,23 +22,97 @@ While the camera component lets you access what your machine's camera sees, the
...
+
+For configuration information, click on the model name:
+
+<!-- prettier-ignore -->
+Model | Description <a name="model-table"></a>
+----- | -----------
+[`mlmodel`](./mlmodel/) | A detector or classifier that uses a `tensorflow-lite` model available on the machine’s hard drive to draw bounding boxes around objects or returns a class label and confidence score.
+[`color_detector`](./color_detector/) | A heuristic detector that draws boxes around objects according to their hue (does not detect black, gray, and white).
+[`obstacles_pointcloud`](./obstacles_pointcloud/) | A segmenter that identifies well-separated objects above a flat plane.
+[`detector_3d_segmenter`](./detector_3d_segmenter/) | This model takes 2D bounding boxes from an object detector and projects the pixels in the bounding box to points in 3D space.
```
Comment (sguequierre) [inline docs/ml/vision/_index.md:102]:
```suggestion
[`detector_3d_segmenter`](./detector_3d_segmenter/) | A segmenter that takes 2D bounding boxes from an object detector and projects the pixels in the bounding box to points in 3D space.
```
(opt) If that's the case maybe this could help clear up if others were similarly confused?

File: docs/ml/vision/_index.md (hunk 6 above)
Comment (sguequierre) [reply docs/ml/vision/_index.md:94]: Got it, makes sense. And yeah, maybe

**Review by sguequierre — APPROVED** [review]
LGTM one opt comment!

File: docs/ml/vision/_index.md (hunk 16 above)
Comment (npentrel) [reply docs/ml/vision/_index.md:102]: Perfect yep

File: docs/tutorials/services/navigate-with-rover-base.md [hunk 17]
```diff
@@ -600,8 +600,8 @@ Attribute information for an `ultrasonic` [camera](/components/camera/) is the s
 
 {{< /alert >}}
 
-If you want the robot to be able to automatically detect obstacles in front of it, [configure a Vision service segmenter](/ml/vision/segmentation/).
-For example, [configure](/ml/vision/segmentation/#configure-an-obstacles_depth-segmenter) the Vision service model [`obstacles_depth`](/ml/vision/segmentation/#configure-an-obstacles_depth-segmenter) to detect obstacles in front of the robot.
+If you want the robot to be able to automatically detect obstacles in front of it, [configure a Vision service segmenter](/ml/vision/#segmentation).
```
Comment (npentrel) [inline docs/tutorials/services/navigate-with-rover-base.md:603]:
```suggestion
If you want the robot to be able to automatically detect obstacles in front of it, [configure a Vision service segmenter](/ml/vision/#segmentations).
```
